openhtf/core/test_executor.py

```diff
@@ -146,6 +146,12 @@ def _thread_proc(self):
 
       # Everything is set, set status and begin test execution.
       self._execute_test_phases(phase_exec)
+      
+  def _thread_exception(self, exc_type, exc_val, exc_tb):
+    """When the framework throws an exception, turn it into a failure code."""
+    if self.test_state and not self.test_state.is_finalized:
+      self.test_state.test_record.add_outcome_details(exc_type.__name__, exc_val.message)
+    return True
```

Comment:
Agree we should look into this more.

- Looking at `util/threads.py` it seems like this is mainly a matter of logging (if the exception is suppressed, it is not printed or logged, otherwise, it is sent to both logging.exception and stderr) or am I missing another side effect that can occur as a result of the thread raising an exception?
- It seems like returning True here risks silencing some errors completely (except for the outcome detail added to the test record). Errors in plug initialization are logged in PlugManager.initialize_plugs but other errors might be silenced.
- I noticed PhaseExecutorThread has sort of the opposite behavior—it never suppresses exceptions. It also makes its own call to logger.exception, so in the console you get “Phase <...> raised an exception” followed by “Thread raised an exception” followed by “Exception in thread” (three printouts of the same exception). We should see if the logging behavior can be more consistent (and not unnecessarily repetitive?).